tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -118,6 +118,34 @@ static inline int z_vrfy_syscall_arg64(uint64_t arg)
 }
 #include <syscalls/syscall_arg64_mrsh.c>
 
+ /* System call made for test test_system_call_separate_stack */
+int z_impl_syscall_separate_stack(void)
+{
+	/* Test that syscall is running on a separate stack, not accessible
```

Comment:
This might be a bit misleading, we do not test here that the separate stack is not accessible to the calling thread, you only test that the variable is outside the thread's normal stack.

Actually, I guess you could use the API to check that the variable is not user-accessible :)